src/ucp/core/ucp_ep.c

```diff
@@ -1034,16 +1038,118 @@ void ucp_ep_destroy(ucp_ep_h ep)
     return;
 }
 
-int ucp_ep_config_lane_is_equal(const ucp_ep_config_key_t *key1,
-                                const ucp_ep_config_key_t *key2,
-                                ucp_lane_index_t lane, int compare_types)
+static int
+ucp_ep_config_lane_is_proxy_equal(const ucp_ep_config_key_t *key1,
+                                  ucp_lane_index_t lane1,
+                                  const ucp_ep_config_key_t *key2,
+                                  ucp_lane_index_t lane2)
+{
+    ucp_lane_index_t proxy_lane1 = key1->lanes[lane1].proxy_lane;
+    ucp_lane_index_t proxy_lane2 = key2->lanes[lane2].proxy_lane;
+
+    if ((proxy_lane1 == UCP_NULL_RESOURCE) !=
+        (proxy_lane2 == UCP_NULL_RESOURCE)) {
+        /* they have different proxy_lane configurations */
+        return 0;
+    }
+
+    return (/* both lanes are not proxies */
+            (proxy_lane1 == UCP_NULL_RESOURCE) &&
+            (proxy_lane2 == UCP_NULL_RESOURCE))               ||
+           (/* both lanes are proxies for themselves */
+            (proxy_lane1 == lane1) && (proxy_lane2 == lane2)) ||
+           /* both lanes are proxies for the same peer */
+           ucp_ep_config_lane_is_peer_equal(key1, proxy_lane1,
```

Comment:
why does `ucp_ep_config_lane_is_peer_equal` calls `ucp_ep_config_lane_is_proxy_equal`?

Reply:
in order to check that lanes are proxies for the same lanes

Reply:
waiting for the https://github.com/openucx/ucx/pull/5688 being merged

Reply:
done